pybop/problems/fitting_problem.py

```diff
@@ -134,3 +135,112 @@ def evaluateS1(self, inputs: Inputs):
+            for signal in problem.signal:
+                combined_dataset["Time [s]"] = np.concatenate(
+                    (combined_dataset["Time [s]"], problem._time_data)
+                )
+                combined_dataset["Combined signal"] = np.concatenate(
+                    (combined_dataset["Combined signal"], problem._target[signal])
+                )
+
+        super().__init__(
+            parameters=combined_parameters,
+            model=None,
+            signal=["Combined signal"],
+        )
+        self._dataset = combined_dataset.data
+        self.parameters.initial_value()
+
+        # Unpack time and target data
+        self._time_data = self._dataset["Time [s]"]
+        self.n_time_data = len(self._time_data)
+        self.set_target(combined_dataset)
+
+    def evaluate(self, inputs: Inputs):
+        """
+        Evaluate the model with the given parameters and return the signal.
+
+        Parameters
+        ----------
+        inputs : Inputs
+            Parameters for evaluation of the model.
+
+        Returns
+        -------
+        y : np.ndarray
+            The model output y(t) simulated with given inputs.
+        """
+        inputs = self.parameters.verify(inputs)
+        self.parameters.update(values=list(inputs.values()))
+
+        y = {"Combined signal": np.asarray([])}
+        for problem in self.problem_list:
+            problem_inputs = problem.parameters.as_dict()
+            for signal in problem.signal:
+                yi = problem.evaluate(problem_inputs)
+                y["Combined signal"] = np.concatenate(
+                    (y["Combined signal"], yi[signal])
+                )
+
+        return y
+
+    def evaluateS1(self, inputs: Inputs):
+        """
+        Evaluate the model with the given parameters and return the signal and its derivatives.
+
+        Parameters
+        ----------
+        inputs : Inputs
+            Parameters for evaluation of the model.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the simulation result y(t) and the sensitivities dy/dx(t) evaluated
+            with given inputs.
+        """
+
+        inputs = self.parameters.verify(inputs)
+        self.parameters.update(values=list(inputs.values()))
+
+        # y = np.empty((self._target_length))
+        # dy = np.empty((self._target_length, self.n_parameters))
+
+        y = {"Combined signal": np.asarray([])}
+        dy = None
+        for problem in self.problem_list:
+            problem_inputs = problem.parameters.as_dict()
+            for signal in problem.signal:
+                yi, dyi = problem.evaluateS1(problem_inputs)
+                y["Combined signal"] = np.concatenate(
+                    (y["Combined signal"], yi[signal])
+                )
+                if dy is None:
+                    dy = dyi
+                else:
+                    dy = np.concatenate((dy, dyi))
+
+        return (y, dy)
```

Comment:
This completes a `evaluate()` for every output signal; however, we provide all signals for each `evaluate()`. I spent a bit of time thinking that the lower logic statement wasn't used, before realising it's a catch for the first iteration. I workshopped an alternative for you to consider:

```suggestion
        combined_signal = []
        all_derivatives = []
        
        for problem in self.problem_list:
            problem_inputs = problem.parameters.as_dict()
            signal_values, dyi = problem.evaluateS1(problem_inputs)
            
            # Collect signals and derivatives
            for signal in problem.signal:
                combined_signal.extend(signal_values[signal])
            all_derivatives.append(dyi)
        
        y = {"Combined signal": np.asarray(combined_signal)}
        dy = np.concatenate(all_derivatives) if all_derivatives else None
        
        return (y, dy)
```